src/basilisp/lang/keyword.py

```diff
@@ -53,9 +58,15 @@ def __lt__(self, other):
             return False
         return self._ns < other._ns or self._name < other._name
 
-    def __call__(self, m: IAssociative, default=None):
+    def __call__(self, m: Union[IAssociative, IPersistentSet], default=None):
         try:
-            return m.val_at(self, default)
+            if isinstance(m, IPersistentSet):
+                if self in m:
+                    return self
+                else:
+                    return default
+            else:
+                return m.val_at(self, default)
```

Comment:
```suggestion
        if isinstance(m, IPersistentSet):
            return self if self in m else default
        try:
            return m.val_at(self, default)
```

I'd like to try to keep what's in the `try` blocks to a minimum so we're only catching exceptions we intend to catch.

Reply:
right, updated.